prelude/bootstrap.js

```diff
@@ -2020,88 +2056,41 @@ function payloadFileSync(pointer) {
-
-          if (!fs.existsSync(importModulePath)) {
-            throw new Error(
-              `INTERNAL ERROR this file doesn't exist in the virtual file system :${importModulePath}`
-            );
-          }
-          const moduleContent1 = fs.readFileSync(importModulePath);
-          const tmpModulePath1 = path.join(_tmpFolder, moduleName);
-
-          try {
-            fs.statSync(tmpModulePath1);
-          } catch (err) {
-            fs.writeFileSync(tmpModulePath1, moduleContent1, { mode: 0o555 });
-          }
-          return tryImporting(_tmpFolder, e.message);
-        }
+    // Example: moduleFolder = /snapshot/appname/node_modules/sharp/build/Release
+    const modulePkgPathRegex = /.*?node_modules\/((.+?)\/.*)/;
+    // Example: modulePackagePath = sharp/build/Release
+    const modulePackagePath = moduleFolder.match(modulePkgPathRegex)[1];
+    // Example: modulePackageName =  sharp
+    const modulePackageName = moduleFolder.match(modulePkgPathRegex)[2];
+    // Example: modulePkgFolder = /snapshot/appname/node_modules/sharp
+    const modulePkgFolder = moduleFolder.replace(
+      modulePackagePath,
+      modulePackageName
+    );
 
-        // this case triggers on windows mainly.
-        // we copy all stuff that exists in the folder of the .node module
-        // into the temporary folders...
-        const files = fs.readdirSync(moduleFolder);
-        for (const file of files) {
-          if (file === moduleBaseName) {
-            // ignore the current module
-            continue;
-          }
-          const filenameSrc = path.join(moduleFolder, file);
-
-          if (fs.statSync(filenameSrc).isDirectory()) {
-            continue;
-          }
-          const filenameDst = path.join(_tmpFolder, file);
-          const content = fs.readFileSync(filenameSrc);
-
-          fs.writeFileSync(filenameDst, content, { mode: 0o555 });
-        }
-        return tryImporting(_tmpFolder, e.message);
-      }
-    }
     if (insideSnapshot(modulePath)) {
       const moduleContent = fs.readFileSync(modulePath);
 
       // Node addon files and .so cannot be read with fs directly, they are loaded with process.dlopen which needs a filesystem path
       // we need to write the file somewhere on disk first and then load it
+      // the hash is needed to be sure we reload the module in case it changes
       const hash = createHash('sha256').update(moduleContent).digest('hex');
 
-      const tmpFolder = path.join(tmpdir(), hash);
+      // Example: /tmp/<hash>
+      const tmpFolder = path.join(tmpdir(), 'pkg', hash);
       if (!fs.existsSync(tmpFolder)) {
-        fs.mkdirSync(tmpFolder);
+        // here we copy all files from the snapshot module folder to temporary folder
+        // we keep the module folder structure to prevent issues with modules that are statically
+        // linked using relative paths (Fix #1075)
+        fs.mkdirSync(tmpFolder, { recursive: true });
```

Comment:
https://nodejs.org/api/fs.html#fs_fs_mkdirsync_path_options

`recursive` is only supported on Node `v10.12.0` or later. We are still supporting Node v8 at the moment, and technically can't drop it without a major release.

Reply:
Suggestions?

Reply:
You would have to bake it yourself, like what you did with the `copyFolderRecursiveSync`. 

Side note: Node v16.7.0 or later has [fs.cpSync](https://nodejs.org/api/fs.html#fs_fs_cpsync_src_dest_options), which makes those custom functions unnecessary. However, v14 EOL in 2023.

Reply:
Check last commit